src/main/java/servlets/VendorServlet.java

```diff
@@ -33,12 +44,68 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     JsonConverter converter = new JsonConverter();
     String configMap = converter.getConfigMap();
-    response.setContentType("text/html;");
+    response.setContentType(CONTENT_TYPE);
     response.getWriter().println(configMap);
   }
 
-  @Override
-  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // DELETE method --> deletes vendor configuration directory
+  /** Delete the desired Vendor from the fileset and spreadsheets if posible.*/
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+    String vendorID = request.getParameter(VENDOR_ID_PARAM);
+    String deleteStatus = "success";
+
+    // Delete the file from the filesystem and update the sheets.
+    try {
+      deleteVendorFile(vendorID);
+      updateSheets(request);    
+    } catch (IOException e) {
+      deleteStatus = "Could not execute, an IOException was thrown.";
+    } catch (GeneralSecurityException e) {
+      deleteStatus = "Could not execute, a GeneralSecurityException was thrown.";
+    } 
+
+    // Convert the String message into a JSON String.
+    String jsonMessage = messageAsJson(deleteStatus);
+    
+    response.setContentType(CONTENT_TYPE);
+    response.getWriter().println(jsonMessage);
+  }
+
+  private void deleteVendorFile(String vendorID) throws IOException {
+    JsonConverter converter = new JsonConverter();
+    ArrayList<String> vendorIDs = converter.getVendorIDs();
+
+    // Validate that the vendorID exists.
+    if(!vendorIDs.contains(vendorID)) {
+        throw new IOException();
```

Comment:
would a FileNotFoundException be better suited here?

Reply:
Yes, FileNotFoundException would be more specific.